dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/DefaultProgramIndicatorService.java

```diff
@@ -146,6 +147,10 @@ public class DefaultProgramIndicatorService implements ProgramIndicatorService {
 
   private final Cache<String> analyticsSqlCache;
 
+  private final SqlBuilder sqlBuilder;
+
+  public static ImmutableMap<Integer, ExpressionItem> PROGRAM_INDICATOR_ITEMS;
```

Comment:
It is a little bit of an anti-pattern to initialize `static` variables in constructors in singletons. Since this is a Spring singleton bean, the constructor is guaranteed to be invoked once. It is more natural to make it consistent with other dependencies, and instead use this:

```
private final Map<Integer, ExpressionItem> programIndicatorItems;
```

Reply:
Sorry, my bad, this part is actually no longer like this. The problem is that this `PROGRAM_INDICATOR_ITEMS` variable is only used by Unit Tests and not by the runtime. But it can no longer be static, since we need to pass the SqlBuilder. Fixing the unit tests, this will change too.

Reply:
Understand. Yes good to get it out of the implementation if it is only used in tests.